inspirehep/modules/theme/jinja2filters.py

```diff
@@ -668,3 +669,28 @@ def format_date(datetext):
             return date
         elif len(datestruct) == 1:
             return datestruct[0]
+
+
+@blueprint.app_template_filter()
+def is_external_link(url):
+    """Checks if given url starts with the following."""
+    return not url.endswith(('.pdf', '.png', '.jpg', '.jpeg'))
+
+
+@blueprint.app_template_filter()
+def is_institute(institute):
+    """Checks if given url starts with the following."""
+    if institute.lower() in ['kekscan', 'ads', 'cds', 'hal', 'msnet', 'msnet']:
+        return True
+    return False
+
+
+@blueprint.app_template_filter()
+def weblinks(description):
+    """..."""
+    weblinks = open(os.path.join('inspirehep', 'weblinks.kb'), 'r')
```

Comment:
You are right to doubt this method as it would read and parse the file upon every call to this filter. Not great for templating speed. So we'd want to avoid relying on reading arbitrary files every time we render templates. 

Now, since invenio-knowledge is probably not gonna be ported as is. One suggestion is to read the file upon extension initialization in (E.g. `theme/ext.py`) and keeping it in memory as it is only ~500 entries. See one example of states here: https://github.com/inveniosoftware/invenio-access/blob/master/invenio_access/ext.py or https://github.com/inspirehep/invenio-workflows/blob/invenio3/invenio_workflows/ext.py

However, we might have more files like this so we should see how to best implement it see e.g. #927